app/src/sharedTest/java/org/oppia/app/home/HomeActivityTest.kt

```diff
@@ -85,67 +99,97 @@ class HomeActivityTest {
     Intents.release()
   }
 
+  private fun getApplicationDependencies() {
+    launch(HomeInjectionActivity::class.java).use {
+      it.onActivity { activity ->
+        oppiaClock = activity.oppiaClock
+      }
+    }
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerHomeActivityTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
   @Test
-  fun testHomeActivity_firstOpen_hasWelcomeString() {
-    launch(HomeActivity::class.java).use {
+  fun testHomeActivity_recyclerViewIndex0_withProfileId0_displayProfileName_profileNameDisplayedSuccessfully() {
+    launch<HomeActivity>(createHomeActivityIntent(0)).use {
       onView(
         atPositionOnView(
           R.id.home_recycler_view,
           0,
-          R.id.welcome_text_view
+          R.id.profile_name_textview
         )
-      ).check(matches(withText("Welcome to Oppia!")))
+      ).check(matches(withText("Sean!")))
     }
   }
 
   @Test
-  fun testHomeActivity_secondOpen_hasWelcomeBackString() {
-    simulateAppAlreadyOpened()
+  fun testHomeActivity_recyclerViewIndex0_displayGreetingMessageBasedOnTime_goodMorningMessageDisplayedSuccessful() {
+    getApplicationDependencies()
+    oppiaClock.setCurrentTimeMs(1579666500000)
```

Comment:
Let makes this code cleaner. At the top of this file, create this as constant:

`private const val MORNING_TIMESTAMP = 1579666500000 `

and use `MORNING_TIMESTAMP` at this place. Similarly in other cases too.

Reply:
DONE.